src/sph_geom/mod.rs

```diff
@@ -184,120 +183,91 @@ impl Polygon {
-            // Case A.1.a: first solution
-            let delta_root_sq = delta.sqrt();
-            let y1 = (-b - delta_root_sq) / two_a;
-            let x = -(n.y() * y1 + n.z() * z) / n.x();
-
-            let p1 = UnitVect3::new_unsafe(x, y1, z);
-
-            // If it lies on the great circle arc defined by pa and pb, this is the "good one" to return
-            if dot_product(&cross_product(&p1, pa), &cross_product(&p1, pb)) < 0.0 {
-              return Some(p1);
-            } else {
-              // Otherwise return the other one
-              let y2 = (-b + delta_root_sq) / two_a;
-              let x = -(n.y() * y2 + n.z() * z) / n.x();
-
-              return Some(UnitVect3::new_unsafe(x, y2, z));
-            }
-          // Case A.2: there are one solution
-          } else if delta == 0.0 {
-            let y = -b / two_a;
-            let x = -(n.y() * y + n.z() * z) / n.x();
-            return Some(UnitVect3::new_unsafe(x, y, z));
-          } else {
-            // No real solutions
-            return None;
-          }
-        // Case B: Nx == 0
-        } else {
-          // Case B.1: Ny = 0
-          if n.y() == 0.0 {
-            // Case B.1.a: N is pointing towards ez. Only the great circle of lat = 0 can fall in that case.
-            // Therefore a solution can only be found if pa and pb lies on the equator too (lat = 0)
-            if z == 0.0 && z == pa.z() {
-              return Some(UnitVect3::new_unsafe(pa.x(), pa.y(), pa.z()));
-            } else {
-              return None;
-            }
-          // Case B.2: Ny != 0
-          } else {
-            let yn2 = n.y().pow2();
-            let zn2 = n.z().pow2();
-
-            let (x, y) = ((lat_cos.pow2() - zn2 * z2 / yn2).sqrt(), -n.z() * z / n.y());
-
-            let p = UnitVect3::new_unsafe(x, y, z);
-            if dot_product(&cross_product(&p, pa), &cross_product(&p, pb)) < 0.0 {
-              return Some(p);
-            } else {
-              return Some(UnitVect3::new_unsafe(-x, y, z));
-            }
+      if great_circle_arcs_are_overlapping_in_lon(a, b, pa, pb) {
+        let ua = dot_product(a, cross_prod);
+        let ub = dot_product(b, cross_prod);
+        if polygon_edge_intersects_great_circle(ua, ub) {
+          if let Some(intersect) = intersect_point_in_polygon_great_circle_arc(a, b, pa, pb, ua, ub)
+          {
+            vertices.push(intersect);
           }
         }
       }
-
-      left = right;
+      left = right
     }
+    vertices
+  }
+
+  /// Returns `true` if an edge of the polygon intersects the great-circle arc defined by the
+  /// two given points (we consider the arc having a length < PI).
+  pub fn is_intersecting_great_circle_arc(&self, a: &Coo3D, b: &Coo3D) -> bool {
+    self.intersect_great_circle_arc(a, b).is_some()
+  }
 
+  /// Returns the first intersection of a small-circle with the polygon
+  pub fn intersect_small_circle_arc(&self, lat: f64) -> Option<UnitVect3> {
```

Comment:
I realized z accepted by special_points_finder::intersect_small_circle is in 3d coo (so between -1 and 1). I think it is better to ask for latitude in rad instead for this method, what do you think ?

Reply:
I am against the remaining of 'parallel' by 'small_circle': 
'parallel' is more specific thatn 'small_circle', see e.g. [this article](https://en.wikipedia.org/wiki/Circle_of_latitude).

Reply:
I see, that you are re-using the name which is in 'special_point_finder'... I think we should leave 'paralle' here (and change the name in 'special_point_finder', maybe not in this PR?)

Reply:
Have you evaluated the difference between `special_point_finder::intersect_small_circle` and `sph_geom::intersect_parallel`? It seems that I implemented twice the same thing (but in different ways?)...

Reply:
intersect_parallel was before the apparition of special_point_finder. I think this handles the same thing, the solution of a second degree equation (but I may be wrong?). At least for the use case of aladin lite, special_point_finder seems to cover it.

Reply:
Yes so I do not understand why the special_point_finder is called intersect_small_circle and not intersect_parallel because not all small circles are defined by z=cte ? Should we all rename them intersect_parallel ?